Custom scopes

NarekManukyan opened the pull request:
Adds a user configurable way to define scope when using LinkedInUserWidget and LinkedInAuthCodeWidget. This leaves the default scope as r_emailaddress%20r_liteprofile, but allows user to set whatever scopes they want.

dzab5 commented:
can we approve this review?

d3xt3r2909 commented:
Please take a look into other PR it's a way how to resolve it

d3xt3r2909 commented:
What is diff between your PR and this one https://github.com/d3xt3r2909/linkedin_login/pull/84


@NarekManukyan ?

NarekManukyan commented:
1. Using `enum` instead of class for scopes, which is more convenient to use for such cases
2. All scopes available for Linkedin.com are defined.
3. Using `Uri` instead of `String` to parse the final `url` to open the authorization flow.
4. Updated libraries
5. Updated `example`

d3xt3r2909 commented:
Steps 4. And 5 should be anyhow in separate PR. 

Step 1 enum is not flexible enough to be able to give to client possibility to have custom scopes if they needed for some reason. Many scopes that are given in this PR, parsing mechanism is not supporting it. 

Step 3 seems also candidate for other PR since it's different thing from initial problem.

There is no single test inside this PR.

I would suggest to close this PR and if needed to upgrade other one that i mentioned

NarekManukyan commented:
@d3xt3r2909  LinkedIn has a certain list of scopes, which means that flexibility in this matter is not required, on the contrary, you need to have a list that can be updated over time

d3xt3r2909 commented:
We cannot support any of scopes that are already not implemented in this library already :] reason is that linkedin is not giving me access to their marketing dev scope .. so BCS of this I wouldn't put enum bcs that means that this library is supporting scope but actually it doesn't. Commenting is not an option

NarekManukyan commented:
I meant that only those scopes that we support can remain in the enum )

NarekManukyan commented:
Let's close and update #84

d3xt3r2909 commented:
Closing as mentioned above
